modules/programs/zsh/default.nix

```
@@ -412,15 +439,15 @@ in
 
         (mkIf (dotDirAbs != homeDir) {
           home.file."${dotDirRel}/.zshenv".text = ''
-            export ZDOTDIR=${dotDirAbs}
+            ${config.lib.zsh.export "ZDOTDIR" dotDirAbs}
           '';
 
           # When dotDir is set, only use ~/.zshenv to source ZDOTDIR/.zshenv,
           # This is so that if ZDOTDIR happens to be
           # already set correctly (by e.g. spawning a zsh inside a zsh), all env
           # vars still get exported
           home.file.".zshenv".text = ''
-            source ${dotDirAbs}/.zshenv
+            source "${dotDirAbs}/.zshenv"
```

I think this is better:

```suggestion
            source ${lib.escapeShellArg dotDirAbs}/.zshenv
```

Or

```suggestion
            source ${lib.escapeShellArg "${dotDirAbs}/.zshenv"}
```